src/outputs/intram_by_itl.py

```diff
@@ -0,0 +1,144 @@
+"""The main file for the UK Intram by ITL 1 and 2 output."""
+
+# Standard Library Imports
+import logging
+import pathlib
+import os
+from datetime import datetime
+from typing import Callable, Dict, Any, Union
+
+# Third Party Imports
+import pandas as pd
+
+# Local Imports
+import src.outputs.map_output_cols as map_o
+
+OutputMainLogger = logging.getLogger(__name__)
+
+
+def save_detailed_csv(
+        df: pd.DataFrame,
+        dir: Union[pathlib.Path, str],
+        title: str,
+        run_id: int,
+        write_csv: Callable,
+        overwrite: bool = True
+        ): 
+    """Save a df as a csv with a detailed filename.
+
+    Args:
+        df (pd.DataFrame): The dataframe to save
+        dir (Union[pathlib.Path, str]): The directory to save the dataframe to.
+        title (str): The filename to save the df as (excluding data, run id).
+        run_id (int): The current run ID.
+        write_csv (Callable): A function to write to a csv file.
+        overwrite (bool, optional): Whether or not to overwrite any current 
+            files saved under the same name. Defaults to True.
+
+    Raises:
+        FileExistsError: An error raised if overwrite is false and the file
+            already exists.
+
+    """
+    date = datetime.now().strftime("%Y-%m-%d")
+    save_name = f"{title}_{date}_v{run_id}.csv"
+    save_path = os.path.join(dir, save_name)
+    if not overwrite and os.path.exists(save_path):
+        raise FileExistsError(
+            f"File '{save_path}' already exists. Pass overwrite=True if you "
+            "want to overwrite this file."
+        )
+    write_csv(save_path, df)
+
+
+def rename_itl(df: pd.DataFrame, itl: int) -> pd.DataFrame:
+    """Rename ITL columns in a dataframe.
+
+    Args:
+        df (pd.DataFrame): The dataframe containing the ITL columns.
+        itl (int): The ITL level.
+
+    Returns:
+        pd.DataFrame: A df with the renamed ITL columns.
+    """
+    renamer = {
+        f"ITL{itl}21CD": f"Area Code (ITL{itl})",
+        f"ITL{itl}21NM": f"Region (ITL{itl})",
```

Comment:
I think the 21 here might be 2021, in which case this code might not work in future. Can you research this? 
Is there another way of doing this, for example using a regex to pick out the CD or NM?
https://geoportal.statistics.gov.uk/maps/ons::international-territorial-level-itl-geography-hierarchy-boundaries-january-2021-uk-1/about

Reply:
If there are multiple years worth of itl columns, e.g., using two years to compare recent data with older data, a regex could present a problem. 

Will look into this.

Reply:
I've added in a regex that assumed future ITL regions will follow suit (ITL{itl number}{year e.g. 21}CD). This is however not very useful until the source gets updated to newer itl's since ITL region codes are hard coded on the assumption they remain the same throughout the project. I will leave this feature in however since it will be useful in the future